File: Android/lab-1/app/src/main/java/com/example/lab_1/MainActivity.java
package com.example.lab_1;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private EditText greetingEditText;
    private TextView resultTextView;
    private static final int REQUEST_CODE = 1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        greetingEditText = findViewById(R.id.greetingEditText);
        resultTextView = findViewById(R.id.resultTextView);
        Button goToSecondButton = findViewById(R.id.goToSecondButton);

        goToSecondButton.setOnClickListener(v -> {
            String greeting = greetingEditText.getText().toString();

            Intent intent = new Intent(MainActivity.this, SecondActivity.class);
            intent.putExtra("GREETING_TEXT", greeting);
            startActivityForResult(intent, REQUEST_CODE);
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == REQUEST_CODE && resultCode == RESULT_OK) {
            String name = data.getStringExtra("USER_NAME");
            String greeting = greetingEditText.getText().toString();
            resultTextView.setText(greeting + ", " + name);
        }
    }
}

File: Android/lab-2/app/src/main/java/com/example/lab_2/MainActivity.kt
package com.example.lab_2

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val recyclerView: RecyclerView = findViewById(R.id.recyclerView)

        val animals = listOf(
            Animal("Lion", R.drawable.lion),
            Animal("Tiger", R.drawable.tiger),
            Animal("Elephant", R.drawable.elephant),
            Animal("Giraffe", R.drawable.giraffe),
            Animal("Panda", R.drawable.panda),
            Animal("Kangaroo", R.drawable.kangaroo),
            Animal("Zebra", R.drawable.zebra),
            Animal("Penguin", R.drawable.penguin),
            Animal("Koala", R.drawable.koala),
            Animal("Dolphin", R.drawable.dolphin)
        )

        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = AnimalAdapter(animals)
    }
}
